src/query/Querent.js

```diff
@@ -771,6 +798,38 @@ export class Querent {
     return this.q_.all(promises).then(handleCombinedQueryResult_);
   }
 
+  /**
+   * Extend the given bbox with the queryIconPosition values.
+   * @param {!number[]} queryIconPosition The values in px to buffer the bbox.
+   * @param {number} resolution The map view resolution to define the px size correctly.
+   * @param {!import("ol/extent.js").Extent} bbox The bbox to extend.
+   * @return {!import("ol/extent.js").Extent} The extended bbox or null if the queryIconPosition param
+   * is not valid.
+   * @private
+   */
+  extendBboxWithQueryIconPosition_(queryIconPosition, resolution, bbox) {
+    const buffers = queryIconPosition.map((value) => value * resolution);
+    const length = buffers.length;
+    if (!length || length > 4) {
+      // Bad format.
+      return null;
```

Comment:
Might log something here.
What if lengh == 3 ? We will get an bad index error some lines below

Reply:
I've added more explicit comments (case length === 3 is case length !==2 and length !==4).
Please take a look at my unit test if you doubt.
I would avoid the log here but I do an assert out of this function to check if the result is null.